### ISSUE TITLE
Optional field missing in the generated docs

### ISSUE THREAD

**Issue by oeway** [bioimage-io/spec-bioimage-io#378]
In the genearated docs, `optional` is missing for `attachments`: https://github.com/bioimage-io/spec-bioimage-io/blob/gh-pages/rdf_spec_latest.md

In addition, I think we should add `required` for every field that is not optional, it will make the docs easier to read, especially when searching for some certain keyword, it's not intuitive to remember "no optional means required".

**Comment by oeway** [comment]
@FynnBe Could you please look into this when you got some time?

**Comment by FynnBe** [comment]
should be fixed by https://github.com/bioimage-io/spec-bioimage-io/pull/386

**Comment by oeway** [comment]
@FynnBe Thanks for fixing it!

Now, looking at the generated docs in main branch, why there are some fields such as `cite` did not mark as `optional` (nor `required`), is that intended? I think maybe better to make it always explicit.

Another minor suggestion to improve the readablity is to put `option/required` and type to a bracket.

For example:
* <a id="icon"></a>`icon` _optional String_ an icon for the resource

It would be more readable if it's something like this:
* <a id="icon"></a>`icon`: (_optional, String_) an icon for the resource

**Comment by FynnBe** [comment]
> Now, looking at the generated docs in main branch, why there are some fields such as cite did not mark as optional (nor required), is that intended? I think maybe better to make it always explicit.

no, that was not intended. I did want to remove the required/optional from inner types, such as the CiteEntry in `cite`: (required List[CiteEntry]).... I'll try to fix that

> Another minor suggestion to improve the readablity is to put option/required and type to a bracket.

good idea 👍

**Comment by FynnBe** [comment]
fixed/implemented in https://github.com/bioimage-io/spec-bioimage-io/pull/387
